### PR TITLE
Refactor switch statement

### PR DESCRIPTION
**Pull request by alaaelgndy**

### CODE REVIEW

File: actions/actions_decision.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+/*
+type allDecision
+Schema {
+	"command name" : {
+		validation length : execution function
+	}
+}
+*/
+type allDecision map[string]map[int]func() string
```
Comment (xfoxawy) [inline actions/actions_decision.go]: we need better name than allDecisions

File: actions/actions.go [hunk 2]
```diff
@@ -1,330 +1,42 @@
...
-			write(conn, use)
-
-		case "show", "ls":
-			show := data.showHandler()
-			write(conn, show)
-
-		case "bye":
-			data.byeHandler()
+	if len(command) < 1 {
+		write(conn, "please type somthing :D")
```
Comment (xfoxawy) [inline actions/actions.go]: remove emoji, use all CAPITAL letters

File: actions/actions.go [hunk 3]
```diff
@@ -1,330 +1,42 @@
...
+		return
+	}
 
-		default:
-			write(conn, "please use help command to know the commands you can use")
-		}
+	ad := NewDecisionManager(data)
+	checkCommandExist := ad.CheckCommandAvailablity(strings.ToLower(command[0]))
+	if !checkCommandExist {
+		write(conn, "please use help to know our commands")
```
Comment (xfoxawy) [inline actions/actions.go]: use `UNKNOW COMMAND, USE HELP`

File: actions/actions.go [hunk 4]
```diff
@@ -1,330 +1,42 @@
...
+		return
+	}
+	runner := ad.RunCommand(len(command))
```
Comment (xfoxawy) [inline actions/actions.go]: pass the command and check the length inside the method

File: actions/actions_decision.go (hunk 1 above)
Comment (alaaelgndy) [reply actions/actions_decision.go]: CommandsExecuter 
CommandsMapper
Or let me know your opinion

**Pull request merged**